packages/esm-patient-tests-app/src/test-results/trendline/trendline.component.tsx

```diff
@@ -117,6 +117,11 @@ const Trendline: React.FC<TrendlineProps> = ({
 
   const dataset = chartTitle;
 
+  const formatObsDatetime = (obsDatetime: string) => {
+    const isoFormattedString = obsDatetime.replace(' ', 'T').replace('.0', '.000').concat('Z');
+    return isoFormattedString;
+  };
```

Comment:
```suggestion
  const formatObsDatetime = (obsDatetime: string) => {
  return dayjs(obsDatetime).toISOString();
};
```
dayjs already provides this conversion to ISO format.

Reply:
The `obsDatetime` parameter is weird. It's an already formatted date string like this: `2025-01-22 05:28:00.0` Following your suggestion rolls back the date by 3 hours (probably because I am on EAT). 

```
obsDatetime - 2025-01-22 05:28:00.0
isoFormattedString - 2025-01-22T05:28:00.000Z 
formattedWithDayJS - 2025-01-22T02:28:00.000Z
```
thanks for highlighting though I have seen a use case for it on the tooltip for accuracy purposes

Reply:
So, yes, because `dayjs` interprets it as a local timestamp (which is what that format claims it is). Instead, you're adding an adjustment so this is treated as if it were in UTC, which is not something we should be deciding here, but on the backend.

Reply:
Agree with @ibacher ... I'm forgetting, when does the server return something like "2025-01-22 05:28:00.0" (ie without specifying time zone).

Reply:
@mogoodrich highly likely that it does not. This is a modification that is most certainly done at the front end, somewhere up the "callstack".

Reply:
The obstree resource @mogoodrich

Reply:
didn't know that @denniskigen

Reply:
We should certainly consider fixing that @amosmachora  @denniskigen , where does the obstree resource live?

Reply:
I'm referring to [this guy](https://github.com/openmrs/openmrs-module-webservices.rest/blob/master/omod-1.9/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_9/ObsTreeResource1_9.java#L45) in the REST-WS module.

Reply:
thanks @denniskigen  !

@ibacher  @dkayiwa  I'm thinking we want to fix this?  specifically, when generating the value_datetime and obs_datetime here:

https://github.com/openmrs/openmrs-module-webservices.rest/blob/master/omod-1.9/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_9/ObsTreeResource1_9.java#L153

... just format in the standard ISO string format instead?

Reply:
@mogoodrich Yep. We do need to make allowance for, e.g., implementations that have been storing time in UTC rather than local time. But, yeah, ideally, all conversations through the API are done in ISO timestamps and the ambiguity should mostly go away.

Reply:
Thanks @ibacher ... isn't that even more of a reason to include an ISO string here?  Stripping off the timezone seems to assume that the server time zone is the facility timezone, and we just want to display that time to the user, regardless of what timezone their client is in.  (We've done that in the past with things like this: https://github.com/openmrs/openmrs-module-uicommons/blob/master/omod/src/main/webapp/resources/scripts/filters/serverDate.js but this has become problematic when the server is in UTC or some timezone that don't correspond to the clinic timezone)